src/tensor_ops/boolean/cuda_kernels.rs

```diff
@@ -0,0 +1,106 @@
+use super::BooleanKernel;
+use crate::prelude::{cuda::CudaArray, *};
+use cudarc::prelude::*;
+
+use std::sync::Arc;
+
+const MODULE_NAME: &str = "/boolean.ptx";
+const PTX_SRC: &'static str = include_str!(concat!(env!("OUT_DIR"), "/boolean.ptx"));
+const ALL_FN_NAMES: [&str; 4] = ["boolean_not", "boolean_and", "boolean_or", "boolean_xor"];
+
+impl Cuda {
+    fn call_binary<S: Shape>(
+        &self,
+        fn_name: &str,
+        lhs: &CudaArray<S, bool>,
+        rhs: &CudaArray<S, bool>,
+    ) -> Result<CudaArray<S, bool>, <Self as HasErr>::Err> {
+        if !self.dev.has_func(MODULE_NAME, fn_name) {
+            self.dev
+                .load_ptx(PTX_SRC.into(), MODULE_NAME, &ALL_FN_NAMES)?;
+        }
+
+        let shape = lhs.shape;
+        let strides = lhs.shape.strides();
+        let numel = shape.num_elements();
+
+        let mut storage = self.dev.take_async(std::vec![false; numel])?;
```

Comment:
Ah so this is where we need the ValidAsZeroBits? Can you add a todo comment in here, and then after we merge this we can open an issue to track it